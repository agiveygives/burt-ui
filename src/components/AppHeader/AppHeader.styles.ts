import styled from 'styled-components';
import { StyledAppHeaderType } from './AppHeader.types';
import DefaultTheme from 'themes';

const StyledAppHeader = styled.div<StyledAppHeaderType>`
  display: grid;
  grid-template-columns: max-content auto max-content;
  grid-column-gap: 20px;
  align-items: center;
  background-color: ${props => props.theme.colors.header};
  padding: 15px 20px;
  border-radius: ${props => props.isRounded ? '0px 0px 20px 20px' : '0px'};
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
`;

StyledAppHeader.defaultProps = {
  theme: DefaultTheme,
  isRounded: false,
};

export default StyledAppHeader;
