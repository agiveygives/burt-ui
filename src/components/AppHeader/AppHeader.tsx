import React, { useState } from 'react';
import classNames from 'classnames';
import Navigation from './Navigation';
import useScrollPosition from '../../hooks/useScrollPosition';
import { AppHeaderPropTypes } from '../AppHeader/types';
import styles from './styles.module.css';

const AppHeader = ({ logoUri, navLinks, menu }: AppHeaderPropTypes) => {
  const scrollPosition = useScrollPosition();

  return (
    <header className={classNames(styles.header, { [`${styles.headerScroll}`]: scrollPosition > 0 })}>
      <div className={styles.logoContainer}>
        <img className={styles.logo} src={logoUri} alt='' />
      </div>

      <Navigation navLinks={navLinks} scrolled={scrollPosition > 0} />

      <div>
        {"menu.icon"}
      </div>
    </header>
  )
};

export default AppHeader;
