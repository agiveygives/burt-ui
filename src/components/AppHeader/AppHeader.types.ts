import { NavigationType } from './Navigation/Navigation.types';

export type StyledAppHeaderType = {
  isRounded?: boolean;
}
export type AppHeaderType = {
  logoSrc: string;
} & NavigationType & StyledAppHeaderType;
