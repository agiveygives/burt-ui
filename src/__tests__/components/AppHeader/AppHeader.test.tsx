import React from 'react';
import { render } from '@testing-library/react';
import AppHeader from '../../../components/AppHeader';

describe('<AppHeader />', () => {
  it('should render without crashing', () => {
    const wrapper = render(<AppHeader
      logoUri='https://cdn.discordapp.com/avatars/313842639084584971/6cd4851ad3f0e86d346ca7132c24a032.webp?size=160'
      navLinks={[
        { href: 'javascript:void(0)', display: 'Dashboard' },
        { href: 'javascript:void(0)', display: 'Shop' },
        { href: 'javascript:void(0)', display: 'About Us' },
      ]}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
