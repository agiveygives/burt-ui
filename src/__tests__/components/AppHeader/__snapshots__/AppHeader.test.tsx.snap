// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHeader /> should render without crashing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="src-components-AppHeader-header-clJtYMJoEg"
      >
        <div
          class="src-components-AppHeader-logoContainer-Yt98Wbcj1d"
        >
          <img
            alt=""
            class="src-components-AppHeader-logo-QJXOYcsaio"
            src="https://cdn.discordapp.com/avatars/313842639084584971/6cd4851ad3f0e86d346ca7132c24a032.webp?size=160"
          />
        </div>
        <ul
          class="src-components-AppHeader-Navigation-container-b3QpzcsypF"
        >
          <li>
            <a
              class="src-components-AppHeader-Navigation-link-ZkbHIOIacv"
              href="javascript:void(0)"
            >
              Dashboard
            </a>
          </li>
          <li>
            <a
              class="src-components-AppHeader-Navigation-link-ZkbHIOIacv"
              href="javascript:void(0)"
            >
              Shop
            </a>
          </li>
          <li>
            <a
              class="src-components-AppHeader-Navigation-link-ZkbHIOIacv"
              href="javascript:void(0)"
            >
              About Us
            </a>
          </li>
        </ul>
        <div>
          menu.icon
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="src-components-AppHeader-header-clJtYMJoEg"
    >
      <div
        class="src-components-AppHeader-logoContainer-Yt98Wbcj1d"
      >
        <img
          alt=""
          class="src-components-AppHeader-logo-QJXOYcsaio"
          src="https://cdn.discordapp.com/avatars/313842639084584971/6cd4851ad3f0e86d346ca7132c24a032.webp?size=160"
        />
      </div>
      <ul
        class="src-components-AppHeader-Navigation-container-b3QpzcsypF"
      >
        <li>
          <a
            class="src-components-AppHeader-Navigation-link-ZkbHIOIacv"
            href="javascript:void(0)"
          >
            Dashboard
          </a>
        </li>
        <li>
          <a
            class="src-components-AppHeader-Navigation-link-ZkbHIOIacv"
            href="javascript:void(0)"
          >
            Shop
          </a>
        </li>
        <li>
          <a
            class="src-components-AppHeader-Navigation-link-ZkbHIOIacv"
            href="javascript:void(0)"
          >
            About Us
          </a>
        </li>
      </ul>
      <div>
        menu.icon
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
